framework/include/executioners/Steady.h

```diff
@@ -45,6 +45,15 @@ class Steady : public Executioner
 
   virtual bool lastSolveConverged() const override { return _last_solve_converged; }
 
+  /**
+   * Get optimization iteration number from main app (when Steady solves an optimization problem)
+   */
+  virtual unsigned int getIterationNumberOutput() const;
+  /**
+   * Set optimization iteration number from main app (when Steady solves an optimization problem)
```

Comment:
same

Reply:
mention the subapp as the current use case

Reply:
These APIs are probably what I like least about the current PR as it exists only for the purpose of the Optimization module ... no framework Steady cases have multiple output iterations (with the exception of per-solve-iterations). Perhaps the exception is the best reason for the existence of these APIs

Reply:
I agree with Alex. @GiudGiud mentioned fixed point iteration solves as a use case for which we may want to have additional, tailored output: Would it make sense to rename that API as something like (get/set)OutputTimeStep and move it up the executioner hierarchy? That may be good if it's going to find more use.

Alternatively, if multiple usage isn't going to happen for that executioner API, we could create a `Steady` executioner class in the optimization module that encapsulates the behavior that the optimization module expects. I think this may not be a great idea since Guillaume thought what we are trying to do here can be extended to other cases.